sdk/cosmos/azure-cosmos/azure/cosmos/aio/_container.py

```diff
@@ -1112,23 +1109,61 @@ async def upsert_item(
         )
         return result
 
+    @distributed_trace_async
+    async def semantic_rerank(
+            self,
+            reranking_context: str,
+            documents: List[str],
+            semantic_reranking_options: Optional[Dict[str, Any]] = None
+    ) -> CosmosDict:
+        """
+        Rerank a list of documents using semantic reranking.
+
+        This method uses a semantic reranker to score and reorder the provided documents
+        based on their relevance to the given reranking context.
+
+        Args:
+            reranking_context (str): The context or query string to use for reranking the documents.
+            documents (List[str]): A list of documents (as strings) to be reranked.
+            semantic_reranking_options (Optional[Dict[str, Any]]): Optional dictionary of additional
+                options to customize the semantic reranking process. Supported options:
+                - **return_documents** (bool): Whether to return the document text in the response. 
```

Comment:
batch size is missing here, should we include this in the args docs since it is in the tests?